File: examples/calculator-python-hexagonal-cqrs/app/application/command/calculate_command_service.py
from app.domain.calculation.model import Calculation, Operator
from app.domain.calculation.repository import CalculationRepository
from app.domain.calculation.value import CalculationId


class CalculateCommandService:
    """Write side: performs a calculation and returns only its ID."""

    def __init__(self, calculation_repository: CalculationRepository) -> None:
        self._calculation_repository = calculation_repository

    def calculate(self, operand_a: float, operator: Operator, operand_b: float) -> CalculationId:
        calculation_id = CalculationId.generate()
        calculation = Calculation.perform(calculation_id, operand_a, operator, operand_b)
        self._calculation_repository.save(calculation)
        calculation.pop_events()
        return calculation_id

File: examples/calculator-java-hexagonal-cqrs/src/main/java/com/oopforge/example/calculator/application/command/CalculateCommandService.java
package com.oopforge.example.calculator.application.command;

import com.oopforge.example.calculator.domain.Calculation;
import com.oopforge.example.calculator.domain.CalculationId;

/**
 * Write side: performs a calculation and returns only its ID.
 */
public class CalculateCommandService implements Calculate {

    private final CalculationRepository calculationRepository;

    public CalculateCommandService(CalculationRepository calculationRepository) {
        this.calculationRepository = calculationRepository;
    }

    @Override
    public CalculationId handle(CalculateCommand command) {
        CalculationId id = CalculationId.generate();
        Calculation calculation = Calculation.perform(id, command.operandA(), command.operator(), command.operandB());
        calculationRepository.save(calculation);
        calculation.popEvents();
        return id;
    }
}
